### PR TITLE
Add sensor counter metrics to TrajectoryCollator class

### PR DESCRIPTION
**Pull request by jkammerl**
Adds metrics to monitor the incoming sensor messages processed by the TrajectoryCollator class.

### CODE REVIEW

File: cartographer/sensor/internal/trajectory_collator.cc [hunk 1]
```diff
@@ -57,5 +63,29 @@ common::optional<int> TrajectoryCollator::GetBlockingTrajectoryId() const {
...
 
+void TrajectoryCollator::RegisterMetrics(
+    metrics::FamilyFactory* family_factory) {
+  collator_metrics_family_ = family_factory->NewCounterFamily(
+      "collator_input_total", "Sensor data received");
+}
+
+metrics::Counter* TrajectoryCollator::GetOrCreateSensorMetric(
+    const std::string& sensor_id, int trajectory_id) {
+  const std::string map_key = sensor_id + "/" + std::to_string(trajectory_id);
```
Comment (gaschler) [inline cartographer/sensor/internal/trajectory_collator.cc]: (Note that "/" would not be allowed in a metric label.)

File: cartographer/sensor/internal/trajectory_collator.cc [hunk 2]
```diff
@@ -57,5 +63,29 @@ common::optional<int> TrajectoryCollator::GetBlockingTrajectoryId() const {
...
+
+  auto metrics_map_itr = metrics_map_.find(map_key);
+  if (metrics_map_itr != metrics_map_.end()) {
+    return metrics_map_itr->second;
+  }
+
+  LOG(INFO) << "Create metrics handler for key: " << map_key;
+  auto new_counter = collator_metrics_family_->Add(
```
Comment (gaschler) [inline cartographer/sensor/internal/trajectory_collator.cc]: auto*

File: cartographer/sensor/internal/trajectory_collator.h [hunk 3]
```diff
@@ -52,7 +54,18 @@ class TrajectoryCollator : public CollatorInterface {
 
   common::optional<int> GetBlockingTrajectoryId() const override;
 
+  static void RegisterMetrics(metrics::FamilyFactory* family_factory);
+
  private:
+  metrics::Counter* GetOrCreateSensorMetric(const std::string& sensor_id,
+                                            int trajectory_id);
+
+  static cartographer::metrics::Family<metrics::Counter>*
```
Comment (gaschler) [inline cartographer/sensor/internal/trajectory_collator.h]: necessary to declare here?

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 1.

**Pull request merged**